packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,129 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
+        }
+    }, [fullScreen, props.showControls]);
+
+    function showControlsHandler(): void {
+        clearTimeout(timeoutRef.current as NodeJS.Timeout);
+
+        setShowControls(true);
+        timeoutRef.current = global.setTimeout(() => {
+            setShowControls(false);
+        }, 5000);
+    }
+
+    const onVideoPressHandler = useCallback((): void => {
+        if (!props.showControls) {
+            return;
+        }
+        if (!showControls) {
+            showControlsHandler();
+        } else {
+            clearTimeout(timeoutRef.current as NodeJS.Timeout);
+            setShowControls(false);
+        }
+    }, [props.showControls, showControls]);
+
+    function fullScreenHandler(isFullScreen: boolean): void {
+        setFullScreen(isFullScreen);
+        StatusBar.setHidden(isFullScreen);
+    }
+
+    const videoProps: VideoProperties = {
+        testID: props.name,
+        source: { uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined },
+        muted: props.muted,
+        repeat: props.loop,
+        controls: props.showControls,
+        onLoadStart: () => setStatus(StatusEnum.LOADING),
+        onError: () => setStatus(StatusEnum.ERROR),
+        useTextureView: false,
+        resizeMode: props.aspectRatio ? "contain" : "stretch",
+        onProgress: ({ currentTime }: OnProgressData) => setCurrentPlayTime(currentTime)
+    };
+
+    const isAndroid = Platform.OS === "android";
+    const errorText = status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load</Text>;
+
     return (
-        <View style={styles.container}>
-            {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
-            {status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load :(</Text>}
-            <Video
-                testID={props.name}
-                source={{ uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined }}
-                paused={!props.autoStart}
-                muted={props.muted}
-                repeat={props.loop}
-                controls={props.showControls}
-                onLoadStart={() => setStatus(StatusEnum.LOADING)}
-                onLoad={data => {
-                    setStatus(StatusEnum.READY);
-                    setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
-                }}
-                onError={() => setStatus(StatusEnum.ERROR)}
-                style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
-                useTextureView={false}
-                resizeMode={props.aspectRatio ? "contain" : "stretch"}
-            />
-        </View>
+        <Fragment>
+            {isAndroid && (
+                <Modal
+                    isVisible={fullScreen}
+                    style={styles.fullScreenVideoPlayer}
+                    onBackButtonPress={() => fullScreenHandler(false)}
```

Comment:
If you utilise `useCallback` here and [here](https://github.com/mendix/widgets-resources/pull/1303/files#diff-5b19cf908530b18852147d085549427ea128e2eb1da3dddad52826adb3d6aae7R131) you can optimise rendering

Reply:
It largely depends on how `Modal` is implemented. Let me know what you think

Reply:
I found it is being passed to the native side as a callback and the default value is `()=>null`, I don't think we should use `useCallback` here, please tell me if you have a different opinion.

Reply:
Sounds good. I think it's only beneficial if `useMemo` is used in `Modal`.